subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -885,18 +970,34 @@ func generateAffinityGroupBindInfo(
 // collectPreemptionVictims collects preemption victims of an affinity group.
 // If any of the GPUs allocated for the whole group is still used by a pod,
 // we will wait for the preemption, as the group is gang-scheduled.
-func collectPreemptionVictims(groupPhysicalPlacement map[int32][]CellList, newGroup bool) (map[string]common.Set, []string) {
+func collectPreemptionVictims(
+	groupPhysicalPlacement map[int32][]CellList,
+	priority CellPriority,
+	groupName string) (map[string]common.Set, []string) {
+
 	preemptionVictims := map[string]common.Set{}
 	var nodesHaveVictims []string
-	if newGroup {
-		for gpuNum := range groupPhysicalPlacement {
-			for podIndex := range groupPhysicalPlacement[gpuNum] {
-				for _, gpu := range groupPhysicalPlacement[gpuNum][podIndex] {
-					pGpu := gpu.(*PhysicalCell)
-					if victimGroup := pGpu.GetAffinityGroup(); victimGroup != nil {
-						// for any victim pod, gang-preempt all the other pods from the same affinity group
-						for _, victims := range victimGroup.allocatedPods {
-							for _, v := range victims {
+	for gpuNum := range groupPhysicalPlacement {
+		for podIndex := range groupPhysicalPlacement[gpuNum] {
+			for _, gpu := range groupPhysicalPlacement[gpuNum][podIndex] {
+				pGpu := gpu.(*PhysicalCell)
+				if victimGroup := pGpu.GetAffinityGroup(); victimGroup != nil && victimGroup.name != groupName {
+					// there are two cases of finding a running pod on the allocated resources:
+					// 1. the running pod is a preemption victim.
+					// 2. the running pod used resource partially released by the current group,
+					// but the group wants to schedule a pod again.
+					// our principle is we allow preemption if the running pod's priority is lower than that
+					// of the group to be scheduled (the 2nd case may also satisfy this condition, and we
+					// allow such preemption). otherwise the running pod cannot be preempted, and the pod
+					// to be scheduled will wait.
+					if pGpu.GetPriority() >= priority {
+						panic(fmt.Sprintf("Allocated resource (%v) has been used by guaranteed group %v",
+							pGpu.GetPhysicalPlacementString(), victimGroup.name))
```

Comment:
Adjust log, such as:

"Resources previously allocated has been allocated to another non-preemptible group %v; pod should wait"